File: oop-alternatives/modules-example.go
// in one file
package a

func DoA() { // public functions start with caps
	...
}

func doB() { // module-private functions start with lowercase
	...
}

// in another
package b

import "a"

a.DoA() // works, because DoA is exported
a.doB() // compiler error!

File: oop-alternatives/modules-example.cpp
// in one file
namespace a {
	void doA() {
		...
	}
	// You know not to call here.
	namespace internal {
		void doB() { 
			...
		}
	}
}

// in another
#include "a.hpp"
namespace b {
	using a;
	a::doA() // works, because doA is exported
	a::doB() // you separated it, so no issues
}
